app/routes/role.routes.js

```diff
@@ -33,7 +33,7 @@ module.exports = function(routes) {
     // delete a role given an id
     routes.delete('/:id', function(request, response) {
         roleController.deleteRoleById(request.params.id)
-            .then(responseHandler.sendJsonResponse(response))
+            .then(responseHandler.sendSuccessfulDeleteJsonResponse(response))
```

Comment:
This has to be wrong. You're invoking the function directly here, which should result in directly sending the response